server/sonar-server/src/main/java/org/sonar/server/view/index/ViewIndexer.java

```diff
@@ -19,51 +19,59 @@
  */
 package org.sonar.server.view.index;
 
+import com.google.common.collect.ImmutableSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
 import org.elasticsearch.action.index.IndexRequest;
 import org.sonar.api.utils.System2;
 import org.sonar.db.DbClient;
 import org.sonar.db.DbSession;
 import org.sonar.db.component.ComponentDto;
 import org.sonar.db.component.UuidWithProjectUuidDto;
-import org.sonar.server.es.BaseIndexer;
 import org.sonar.server.es.BulkIndexer;
 import org.sonar.server.es.EsClient;
+import org.sonar.server.es.IndexTypeId;
+import org.sonar.server.es.StartupIndexer;
 
 import static com.google.common.collect.Maps.newHashMap;
+import static org.sonar.server.view.index.ViewIndexDefinition.INDEX_TYPE_VIEW;
 
-public class ViewIndexer extends BaseIndexer {
+public class ViewIndexer implements StartupIndexer {
 
   private final DbClient dbClient;
+  private final EsClient esClient;
 
   public ViewIndexer(System2 system2, DbClient dbClient, EsClient esClient) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "system2". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1172)